In [15]:
control_conversion = round((1003/100047501) * 100, 4)
control_conversion

0.001

In [16]:
test_conversion = round((1099/100001055) * 100, 4)
test_conversion

0.0011

In [17]:
growth = round(((1099 - 1003) / 1003) * 100, 2)
growth

9.57

По результатам АБ теста мы видим рост платежей в тестовой группе на 9.57% по сравнению с контрольной. Это неплохой показатель, однако нужно отметить, что сама конверсия в тестовой группе выше конверсии контрольной группы лишь на 0.0001%. Это очень низкий показатель, как низки и сами уровни конверсии, особенно для огромных выборок, включающих более 100 млн человек.

Давайте теперь посчитаем p-value и z-score, чтобы оценить статистическую значимость эксперимента.

In [21]:
import numpy as np
from statsmodels.stats.proportion import proportions_ztest
alpha = 0.05
payments = np.array([1003, 1099])
users = np.array([100047501, 100001055])
z_score, p_value = proportions_ztest(payments, users, alternative='two-sided')
print(f'Z-score: {z_score:.4f}')
print(f'P-value: {p_value:.4f}')

Z-score: -2.1046
P-value: 0.0353


In [22]:
if p_value > alpha:
    print(f'p-value ({p_value:.4f}) > alpha ({alpha}). Результат статистически не значим. Не отклоняем нулевую гипотезу. В тестовой группе не произошло изменений конверсии.')
else:
    print(f'p-value ({p_value:.4f}) <= alpha ({alpha}). Результат статистически значим. Отклоняем нулевую гипотезу. В тестовой группе конверсия изменилась.')

p-value (0.0353) <= alpha (0.05). Результат статистически значим. Отклоняем нулевую гипотезу. В тестовой группе конверсия изменилась.


Z-тест на основе наших данных вычислил p-value на уровне 0.0353, что меньше классического уровня статистической значимости. Это означает, что результат является статистически значимым, и формально мы можем отклонить нулевую гипотезу и внедрить изменения.

Тем не менее, ввиду малых абсолютных величин рекомендуется быть очень осторожными с внедрением. Возможно, стоит продлить тест и проверить правильность сбора данных.